Sources/PackageGraph/RepositoryPackageContainer.swift

```diff
@@ -124,6 +104,28 @@ public class RepositoryPackageContainer: BasePackageContainer, CustomStringConve
             currentToolsVersion: currentToolsVersion
         )
     }
+    
+    // Compute the map of known versions.
+    private func knownVersions() throws -> [Version: String] {
+        try self.knownVersionsCache.memoize() {
+            let knownVersionsWithDuplicates = Git.convertTagsToVersionMap(try repository.tags())
+
+            return knownVersionsWithDuplicates.mapValues({ tags -> String in
+                if tags.count == 2 {
+                    // FIXME: Warn if the two tags point to different git references.
+                    return tags.first(where: { !$0.hasPrefix("v") })!
```

Comment:
perhaps an opportunity to fix this as well, does not seem very safe but I dont know the history here

Reply:
Don't know the history here, but I agree, it looks a bit fragile.  That's probably for a separate PR though.  I would think that the list of known versions has already been filtered at this point using some regex that allows `v` as optional but that has already filtered out other tags starting with `v`, so this is probably not an issue in practice.